opentasks/src/main/java/org/dmfs/tasks/TaskListFragment.java

```diff
@@ -497,18 +506,13 @@ private void selectChildView(ExpandableListView expandLV, int groupPosition, int
             {
                 return;
             }
-            // TODO: for now we get the id of the task, not the instance, once we support recurrence we'll have to change that
-            Long selectTaskId = cursor.getLong(cursor.getColumnIndex(Instances.TASK_ID));
 
-            if (selectTaskId != null)
-            {
-                // Notify the active callbacks interface (the activity, if the fragment is attached to one) that an item has been selected.
-
-                // TODO: use the instance URI one we support recurrence
-                Uri taskUri = ContentUris.withAppendedId(Tasks.getContentUri(mAuthority), selectTaskId);
+            // TODO For now we get the id of the task, not the instance, once we support recurrence we'll have to change that, use instance URI that time
+            Uri taskUri = ContentUris.withAppendedId(Tasks.getContentUri(mAuthority), (long) TaskFieldAdapters.TASK_ID.get(cursor));
+            // TODO Should we use TASK_COLOR? (That's not in the projection.)
```

Comment:
No, `TASK_COLOR` should be used as some sort of secondary color, like on a banner, badge or additional color bar.

Reply:
Thanks, good to know. I've updated the parameter names to refer to the task list color.